spec.html

```diff
@@ -903,13 +903,24 @@ <h1>Static Semantics</h1>
-      <p>The mathematical function <emu-eqn id="eqn-abs" aoid="abs">abs(_x_)</emu-eqn> produces the absolute value of _x_, which is <emu-eqn>-_x_</emu-eqn> if _x_ is negative (less than zero) and otherwise is _x_ itself.</p>
-      <p>The mathematical function <emu-eqn id="eqn-min" aoid="min">min(_x1_, _x2_, ..., _xN_)</emu-eqn> produces the mathematically smallest of <emu-eqn>_x1_</emu-eqn> through <emu-eqn>_xN_</emu-eqn>. The mathematical function <emu-eqn id="eqn-max" aoid="max">max(_x1_, _x2_, ..., _xN_)</emu-eqn> produces the mathematically largest of <emu-eqn>_x1_</emu-eqn> through <emu-eqn>_xN_</emu-eqn>. The domain and range of these mathematical functions include *+&infin;* and *-&infin;*.</p>
-      <p>The notation &ldquo;<emu-eqn id="eqn-modulo" aoid="modulo">_x_ modulo _y_</emu-eqn>&rdquo; (_y_ must be finite and nonzero) computes a value _k_ of the same sign as _y_ (or zero) such that <emu-eqn>abs(_k_) &lt; abs(_y_) and _x_-_k_ = _q_ &times; _y_</emu-eqn> for some integer _q_.</p>
-      <p>The mathematical function <emu-eqn id="eqn-floor" aoid="floor">floor(_x_)</emu-eqn> produces the largest integer (closest to positive infinity) that is not larger than _x_.</p>
+      <p>This specification makes reference to two kinds of numeric values:</p>
+      <ul>
+        <li><em>Number</em>: IEEE 754-2008 double-precision floating point values, used as the default numeric type.</li>
+        <li><em>Mathematical value</em>: Arbitrary real numbers, used for specific situations.</li>
+      </ul>
+
+      <p>In the language of this specification, numerical values and operations (including addition, subtraction, negation, multiplication, addition, and comparison) are distinguished among different numeric kinds using subscripts. The subscript <sub><dfn id="𝕗">𝕗</dfn></sub> refers to Numbers, and the subscript <sub><dfn id="𝕧">𝕧</dfn></sub> refers to numeric values. A subscript is used following each numeric value and operation. For brevity, the <sub>𝕗</sub> subscript can be omitted on Number values--a numeric value with no subscript is interpreted to be a Number. An operation with no subscript is interpreted to be a Number operation, unless one of the parameters has a particular subscript, in which case the operation adopts that subscript. For example, 1<sub>𝕧</sub> + 2<sub>𝕧</sub> = 3<sub>𝕧</sub> is a statement about mathematical values, and 1 + 2 = 3 is a statement about Numbers. It is not defined to mix Numbers and mathematical values in either arithmetic or comparison operations, and any such undefined operation would be an editorial error in this specification text.</p>
```

Comment:
numeric values → mathematical values?

Reply:
Also for the 1v + 2v operation, should that be 1v + 2 = 3 instead?

Reply:
> numeric values → mathematical values?

I don't really understand this suggestion; I used the phrase "numeric values" here in order to range over Numbers and mathematical values. Do you have a better word than "numeric" to use here? Note that the BigInt ToNumeric abstract operation also corresponds to this concept.

> Also for the 1v + 2v operation, should that be 1v + 2 = 3 instead?

In the notation in this PR, it's required to put a v subscript on all literal mathematical values. The expression 1v + 2 is undefined, since you can't add a mathematical value and a number. The only implicit subscripts are for operators, not values. Do you see anything I could clarify about the text in this section to explain this better?

Reply:
re numeric values → mathematical values?:
I suspect ljharb is referring to a different occurrence than you're looking at, specifically
`<sub><dfn id="𝕧">𝕧</dfn></sub> refers to numeric values.`
Anyhow, that's the change I made in [this commit](https://github.com/littledan/ecma262/pull/2/commits/1aac62c6bfa40b8d8fe0d6daed4601293461f6cb).

Reply:
This sentence talks about the operation adopting a subscript when one of the operands has a subscript and the other does not. I’m suggesting showing an example of one explicit and one missing subscript, which to my reading means that 1v + 2 is quite defined to be “1v + 2v” - whereas 1v + 2f might be undefined.

Reply:
Earlier in the paragraph, it says `a numeric value with no subscript is interpreted to be a Number`. So `1v + 2` is equivalent to `1v + 2f`, which is a spec error (not defined).

Reply:
Yes - but after that, it says that in the absence of one subscript, the presence of another upgrades the entire operation to match. I’m asking for an example that shows that “adoption”.

Reply:
When you say "entire operation", I think you're thinking of the whole 'expression', but when @littledan says "operation" here, I'm pretty sure he's just talking about (in this example) the plus.

So, roughly speaking, the 'v' subscript on the '1' propagates to the plus, but not to the '2'.

Reply:
Hmm, that seems very strange to have a binary operator not be defined solely based on its two operands

Reply:
Huh? How is plus *not* being defined solely based on its operands?

- If its operands are both mathematical values, it performs a mathematical addition.
- If its operands are both Number values, it performs a Number addition.
- If its operands are anything else, the spec explodes.

Reply:
> So, roughly speaking, the 'v' subscript on the '1' propagates to the plus, but not to the '2'.

If it solely depends on the operates, then why would anything propagate to the plus?

Reply:
> If it solely depends on the operates [operands?], then why would anything propagate to the plus?

I don't really understand the question, because to me those are two ways of saying about the same thing.

Like I said, the "propagates" terminology is roughly speaking. It would be more correct to say that, given an unsubscripted operator, the presence of a v-subscript on an operand allows you to infer a v-subscript for the operator.

Reply:
I was trying to leave the extent of "propagation" really minimal and syntactically obvious.

New plan for omitting subscripts:
- For operations such as +, decide on the type based on the operands.
- For values, do as @jmdyck suggested and omit subscripts on "safe" values, including them on all "unsafe" values. Note that, when we add BigInt, the set of safe values will get smaller, but AFAICT not in a way which affects any of the spec algorithms that I saw.